test/client/contract-client.js

```diff
@@ -396,14 +665,367 @@ describe('readonly calls', () => {
+    const arg = Float64Array.from('42.5')[0]
+    const nodeClient = sinon.stub()
+    nodeClient.readonlySubmit = sinon.stub().withArgs(arg).returns(new Promise((resolve, reject) => {
+      const respXdr = types.ReadonlyResponse()
+      respXdr.fromJSON({
+        result: JSON.stringify(arg * 2),
+        stateStatus: {
+          previousBlock: '3',
+          transactionCount: '1'
+        },
+        status: 1,
+        statusInfo: 'status was good.'
+      })
+      resolve(respXdr)
+    }))
+
+    const client = new ContractClient(testAbi, nodeClient)
+    return client.readonly_f64(arg).then((res) => {
+      expect(res).to.equal(arg * 2)
+    })
+  })
+
+  it('calls readonly_string correctly', () => {
+    const arg = 'abc'
+    const nodeClient = sinon.stub()
+    nodeClient.readonlySubmit = sinon.stub().withArgs(arg).returns(new Promise((resolve, reject) => {
+      const respXdr = types.ReadonlyResponse()
+      respXdr.fromJSON({
+        result: JSON.stringify(arg + arg),
+        stateStatus: {
+          previousBlock: '3',
+          transactionCount: '1'
+        },
+        status: 1,
+        statusInfo: 'status was good.'
+      })
+      resolve(respXdr)
+    }))
+
+    const client = new ContractClient(testAbi, nodeClient)
+    return client.readonly_string(arg).then((res) => {
+      expect(res).to.equal(JSON.stringify(arg + arg))
+    })
+  })
+
+  it('calls readonly_bytes correctly', () => {
+    const arg = Buffer.from('abc')
+    const nodeClient = sinon.stub()
+    nodeClient.readonlySubmit = sinon.stub().withArgs(arg).returns(new Promise((resolve, reject) => {
+      const respXdr = types.ReadonlyResponse()
+      respXdr.fromJSON({
+        result: JSON.stringify(arg + arg),
+        stateStatus: {
+          previousBlock: '3',
+          transactionCount: '1'
+        },
+        status: 1,
+        statusInfo: 'status was good.'
+      })
+      resolve(respXdr)
+    }))
+
+    const client = new ContractClient(testAbi, nodeClient)
+    return client.readonly_bytes(arg).then((res) => {
+      expect(res).to.equal(arg + arg)
+    })
+  })
+
+  it('calls readonly_custom correctly', () => {
+    const nodeClient = sinon.stub()
     const arg = customXDR.MyType()
 
-    nodeClient.readonlySubmit = sinon.fake.returns(new Promise((resolve, reject) => {
+    nodeClient.readonlySubmit = sinon.stub().withArgs(arg).returns(new Promise((resolve, reject) => {
+      const respXdr = types.ReadonlyResponse()
+      respXdr.fromJSON({
+        result: JSON.stringify(arg),
+        stateStatus: {
+          previousBlock: '3',
+          transactionCount: '1'
+        },
+        status: 1,
+        statusInfo: 'status was good.'
+      })
+      resolve(respXdr)
+    }))
+
+    const client = new ContractClient(testAbi, nodeClient, customXDR)
+    return client.readonly_custom(arg.toJSON()).then((res) => {
+      expect(res).to.deep.equal(arg.toJSON())
+    })
+  })
+
+  it('calls function_bool correctly', () => {
```

Comment:
Since these are not readonly calls they should probably be broken out into a separate describe.